### PR TITLE
Clarify/specify/document model card metadata, `model-index`, and pipeline/task types

### PR DESCRIPTION
**Pull request by julien-c**
⚠️ Warning: I got a little carried away with the https://github.com/huggingface/huggingface_hub/pull/265/commits/7d1d7c930cfd0755822e5b39242286480bd31a3d commit (😂), as it may very well break stuff in moon-landing, and it will require others JS-codebase PRs (like @mishig25's) to be rebased.

Let's see if maybe we want to split that change into a different PR

### CODE REVIEW

**Comment by julien-c** [comment]
On further thought the migration path is simple enough that I think this is reviewable/mergeable. Will still tag @mishig25, let me know if i shouldn't merge this when it's reviewed!

**Comment by julien-c** [comment]
CI is flakier than flakes but i've managed to spam it enough that it now passes 🔫 

✅

File: widgets/src/lib/interfaces/Types.ts
Comment (elishowk) [inline widgets/src/lib/interfaces/Types.ts:14]: I think this will break moon-landing code like `moon-landing/server/views/lib/modelInputSnippets.ts` or `server/views/components/InferenceModal/inferenceSnippets.ts` that uses values "as keys". But putting pretty names instead of `key: key` is a good refactoring though

Comment (julien-c) [reply widgets/src/lib/interfaces/Types.ts:14]: yes there will be an associated fix in moon-landing for sue

**Review by LysandreJik — APPROVED** [review]
Thank you for specifying the required/optional nature of elements in the modelcard.

File: docs/hub/adding-a-library.md
Comment (osanseviero) [inline docs/hub/adding-a-library.md:249]: Do we really want to maintain the spec in two places? Maybe we can delete the one below to avoid obsolete content

**Review by osanseviero — APPROVED** [review]
This looks great to me, thanks for this!

File: docs/hub/adding-a-library.md
Comment (julien-c) [reply docs/hub/adding-a-library.md:249]: yes i think it makes sense

Comment (julien-c) [reply docs/hub/adding-a-library.md:249]: i'll leave it for now but we can remove it later

**Pull request merged**
